bugfix: Implement wait for vault readiness to fix bug.

echoboomer opened the pull request:
Signed-off-by: Scott Hawkins <scott@echoboomer.net>

Implements a few new Kubernetes functions for working with resources. I think moving forward this would be a better blueprint for handling similar issues versus parsing flags from the config.

```bash
2023-02-13T20:14 INF internal/k8s/exec.go:34 > Waiting for vault StatefulSet to be created.
2023-02-13T20:14 INF internal/k8s/exec.go:83 > Waiting for StatefulSet vault to be ready. This could take up to 60 seconds.
2023-02-13T20:14 INF internal/k8s/exec.go:97 > All Pods in StatefulSet vault are ready.
LOG: 2023/02/13 20:14:43.625777 /Users/scott/src/kubefirst/internal/k8s/portForward.go:94: Namespace for PF vault
LOG: 2023/02/13 20:14:43.626065 /Users/scott/src/kubefirst/internal/k8s/portForward.go:95: Name for PF vault-0
2023-02-13T20:14 INF internal/k8s/wrappers.go:169 > port forwarding is ready to get traffic
2023-02-13T20:14 INF internal/k8s/wrappers.go:172 > Pod "vault-0" at namespace "vault" has port-forward accepting local connections at port 8200
```

jarededwards commented on cmd/civo/civo.go:
```diff
@@ -375,15 +375,15 @@ func runCivo(cmd *cobra.Command, args []string) error {
-	// todo fix this hack, but vault is unsealed by default in current state
-	time.Sleep(time.Second * 15)
+	// Wait for Vault StatefulSet Pods to transition to Running
+	vaultStatefulSet, err := k8s.ReturnStatefulSetObject(configs.GetCivoConfig().KubeConfigPath, "vault", "vault", 60)
```
```suggestion
	vaultStatefulSet, err := k8s.ReturnStatefulSetObject(kubeconfigPath, "vault", "vault", 60)
```

echoboomer commented:
Implemented recommended changes to references kubeconfig. Also changed out hold for ArgoCD:

```bash
2023-02-14T09:28 INF internal/k8s/exec.go:91 > Waiting for argocd StatefulSet to be created.
2023-02-14T09:28 INF internal/k8s/exec.go:189 > Waiting for argocd-application-controller StatefulSet to be ready. This could take up to 90 seconds.
2023-02-14T09:28 INF internal/k8s/exec.go:203 > All Pods in StatefulSet argocd-application-controller are ready.
LOG: 2023/02/14 09:28:45.905406 /Users/scott/src/kubefirst/internal/k8s/portForward.go:94: Namespace for PF argocd
LOG: 2023/02/14 09:28:45.905504 /Users/scott/src/kubefirst/internal/k8s/portForward.go:95: Name for PF argocd-server-67f9dbb459-2zszq
2023-02-14T09:28 INF internal/k8s/wrappers.go:169 > port forwarding is ready to get traffic
2023-02-14T09:28 INF internal/k8s/wrappers.go:172 > Pod "argocd-server" at namespace "argocd" has port-forward accepting local connections at port 8080

2023-02-14T09:28 INF pkg/helpers.go:676 > port-forward to argocd is available at http://localhost:8080
2023-02-14T09:28 INF pkg/helpers.go:676 > Setting argocd username and password credentials
2023-02-14T09:28 INF pkg/helpers.go:676 > argocd username and password credentials set successfully
2023-02-14T09:28 INF pkg/helpers.go:676 > Getting an argocd auth token
2023-02-14T09:29 INF internal/argocd/argocd.go:374 > requesting auth token from argocd: attempt 1 of 20
2023-02-14T09:29 INF pkg/helpers.go:676 > argocd admin auth token set
2023-02-14T09:29 INF pkg/helpers.go:676 > applying the registry application to argocd
2023-02-14T09:29 INF pkg/shell.go:29 > OUT: application.argoproj.io/registry created

2023-02-14T09:29 INF pkg/shell.go:30 > Command: /Users/scott/.k1/tools/kubectl
2023-02-14T09:29 INF internal/k8s/exec.go:91 > Waiting for vault StatefulSet to be created.
2023-02-14T09:29 INF internal/k8s/exec.go:189 > Waiting for vault StatefulSet to be ready. This could take up to 60 seconds.
2023-02-14T09:29 INF internal/k8s/exec.go:203 > All Pods in StatefulSet vault are ready.
LOG: 2023/02/14 09:29:45.908798 /Users/scott/src/kubefirst/internal/k8s/portForward.go:94: Namespace for PF vault
LOG: 2023/02/14 09:29:45.908907 /Users/scott/src/kubefirst/internal/k8s/portForward.go:95: Name for PF vault-0
2023-02-14T09:29 INF internal/k8s/wrappers.go:169 > port forwarding is ready to get traffic
2023-02-14T09:29 INF internal/k8s/wrappers.go:172 > Pod "vault-0" at namespace "vault" has port-forward accepting local connections at port 8200
```

This now works for both.
